src/azul/service/storage_service.py

```diff
@@ -155,15 +169,19 @@ def upload_multipart_part(self,
 
     def complete_multipart_upload(self,
                                   upload: MultipartUpload,
-                                  etags: Sequence[str]) -> None:
+                                  etags: Sequence[str],
+                                  *,
+                                  overwrite: bool = True,
+                                  ) -> None:
         parts = [
             {
                 'PartNumber': index + 1,
                 'ETag': etag
             }
             for index, etag in enumerate(etags)
         ]
-        upload.complete(MultipartUpload={'Parts': parts})
+        upload.complete(MultipartUpload={'Parts': parts},
```

Comment:
That same exception needs to be handled here as well.

Additionally, we don't want to create a big MP upload only to then realize at the end that the object already exists. The proper way to deal with this for large files is to check explicitly with HeadObject. I don't know if PutObject with If-None-Match fails early before processing the entire request body or after. Unless you can find documentation about this, please add the HeadObject for small files as well.